### ISSUE TITLE
pathlib triggers division w/o __future__ statement

### ISSUE THREAD

**Issue by pmav99** [PyCQA/pylint#2808]
### Steps to reproduce
Create `/tmp/foo.py`
``` python
import pathlib

tmp = pathlib.Path("/tmp")
tmp / "asdf.txt"
```
Check with
```
pylint --reports=no --disable=all --enable=old-division /tmp/foo.py
```

### Current behavior
A warning is issued for division
```
************* Module foo
/tmp/foo.py:4: [W1619(old-division), ] division w/o __future__ statement
```

### Expected behavior
Not sure how easy it is, but it would be nice if pylint could recognise pathlib objects

### pylint --version output
```
$ pylint --version
pylint 2.3.1
astroid 2.2.5
Python 3.7.2 (default, Jan 10 2019, 23:51:51) 
[GCC 8.2.1 20181127]
```

**Comment by PCManticore** [comment]
Thanks for reporting this @pmav99 This is fixed in `master` and It's going to be part of pylint 2.4 when that will be released.

**Comment by asottile** [comment]
this is awesome, looking forward to a release with this.

It also fixes this situation which is pretty nice:

```python
# used to trigger old-division in pylint 2.3.1, now doesn't!
x / float(y)
```

**Comment by asottile** [comment]
oh, there's a regression though:

```
a / int(b)
```

this should trigger `old-division` but now does not
